### DISCUSSION TITLE
NPCOperateItemAction Operate="false" action doesn't do anything

### DISCUSSION THREAD

**Discussion by dakkhuza** [FakeFishGames/Barotrauma#14216] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

When using `operate="false"` to make the npc stop operating whatever they were told to the code mistakenly checks the [OperateTarget of all AIObjectiveOperateItem](https://github.com/FakeFishGames/Barotrauma/blob/85e5d806e6aed75380d5ef6137ae196a519c8b4d/Barotrauma/BarotraumaShared/SharedSource/Events/EventActions/NPCOperateItemAction.cs#L89). 

The issue here is that [NPCOperateItemAction never actually sets an operate target](https://github.com/FakeFishGames/Barotrauma/blob/85e5d806e6aed75380d5ef6137ae196a519c8b4d/Barotrauma/BarotraumaShared/SharedSource/Events/EventActions/NPCOperateItemAction.cs#L76), so there is never anything to check against, thus making it impossible to stop an NPCOperateItemAction using the `operate="false"` property.

It should instead use the item property of the targeted component, e.g. `operateItemObjective.Component.Item == target` instead of `operateItemObjective.OperateTarget == target`

### Reproduction steps

1. Download the example mod
2. Load up the "NPCOperateAction" sub in the sub editor
3. Type `triggerevent npcoperateaction_example` 
4. Observe that the NPC does not cancel the operate objective when instructed and start following the player

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.8.0 (Summer Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by dakkhuza** [comment]
[NPCOperateAction.zip](https://github.com/user-attachments/files/15995064/NPCOperateAction.zip)
Example mod for testing

Expected behavior is:
1. NPC walks over to fabricator and operates it
2. After some time, the NPC stops operating the fabricator
3. The NPC starts following the player

Actual behavior:
NPC is stuck operating the fabricator forever

**Comment by dakkhuza** [comment]
PR fixing this issue by implementing `operateItemObjective.OperateTarget == target` -> `operateItemObjective.Component.Item == target`

This could also be fixed by actually assigning the `OperateTarget` in the objective constructor, however I didn't want to change how the action operated and potentially cause unexpected compatibility issues.

https://github.com/FakeFishGames/Barotrauma/pull/14217

**Comment by Regalis11** [comment]
Thank you for the report, and for taking the time to investigate this! :) I've merged the PR to our private repo now, closing this report.
